src/moscot/problems/space/_mixins.py

```diff
@@ -432,19 +440,46 @@ def correlate(  # type: ignore[misc]
         if sp.issparse(gexp_sc):
             gexp_sc = gexp_sc.toarray()
 
-        corrs = {}
+        corrs = {}  # type: ignore
```

Comment:
```suggestion
        corrs: dict[str,Union[pd.Series,dict[str,pd.Series]] = {}
```

Reply:
this might work. Would try to type it correctly

Reply:
done.